docs/dev/lwt-implementation.md

```diff
@@ -0,0 +1,166 @@
+class Proposal:
+    number: int
+    value: Optional[str] = None
+
+class Acceptor:
+    def __init__(self):
+        self.promised_number = None  # Highest prepare number promised
+        self.accepted_proposal = None  # Accepted proposal (number, value)
+
+    def on_prepare(self, proposal_number: int) -> Tuple[bool, Optional[Proposal]]:
+        if self.promised_number is None or proposal_number > self.promised_number:
+            self.promised_number = proposal_number
+            return True, self.accepted_proposal  # Promise granted
+        return False, None  # Promise denied
+
+    def on_accept(self, proposal: Proposal) -> bool:
+        if self.promised_number is None or proposal.number >= self.promised_number:
+            self.promised_number = proposal.number
+            self.accepted_proposal = proposal
+            return True  # Accepted
+        return False  # Rejected
+
+class Proposer:
+    def __init__(self, proposer_id: int, acceptors: List[Acceptor]):
+        self.proposer_id = proposer_id
+        self.acceptors = acceptors
+        self.proposal_number = proposer_id  # Start with unique ID-based number
+        self.quorum = len(acceptors) // 2 + 1  # Majority
+
+    def prepare(self, value: str) -> bool:
+        self.proposal_number += 10  # Increment to ensure uniqueness (simplified)
+        promises = 0
+        highest_accepted = None
+
+        for acceptor in self.acceptors:
+            promise, accepted_proposal = acceptor.on_prepare(self.proposal_number)
+            if promise:
+                promises += 1
+                if accepted_proposal and (highest_accepted is None or 
+                                         accepted_proposal.number > highest_accepted.number):
+                    highest_accepted = accepted_proposal
+
+        if promises >= self.quorum:
+            # If we have a majority, pick value: use highest accepted or our own
+            self.value = highest_accepted.value if highest_accepted else value
+            return True
+        return False
+
+    def accept(self) -> bool:
+        """Phase 2: Send accept requests with chosen value."""
+        accepts = 0
+        proposal = Proposal(self.proposal_number, self.value)
+
+        for acceptor in self.acceptors:
+            if acceptor.on_accept(proposal):
+                accepts += 1
+
+        return accepts >= self.quorum  # Consensus if majority accepts
+
+class Learner:
+    def __init__(self):
+        self.learned_value = None
+
+    def learn(self, value: str):
+        self.learned_value = value
+        print(f"Learned consensus value: {value}")
+
+def run_paxos():
+    acceptors = [Acceptor() for _ in range(5)]  # 5 acceptors
+    proposer = Proposer(proposer_id=1, acceptors=acceptors)
+    learner = Learner()
+
+    # Proposer tries to propose a value
+    if proposer.prepare("ValueA"):
+        if proposer.accept():
+            learner.learn(proposer.value)
+
+if __name__ == "__main__":
+    run_paxos()
+```
```

Comment:
Do we really need to describe classical paxos instead of referring to the paper. I doubt we can do it better and understanding the paper is hard enough.